src/app-layer-enip-common.c

```diff
@@ -0,0 +1,904 @@
+    //         node->request.path_size);
+
+    DecodeCIPRequestPathPDU(input, input_len, node, offset);
+
+    offset += path_size * sizeof(uint16_t); //move offset past pathsize
+
+    //list of CIP services is large and can be vendor specific, store CIP service  anyways and let the rule decide the action
+    switch (service)
+    {
+        case CIP_RESERVED:
+            SCLogDebug("DecodeCIPRequest - CIP_RESERVED\n");
+            break;
+        case CIP_GET_ATTR_ALL:
+            SCLogDebug("DecodeCIPRequest - CIP_GET_ATTR_ALL\n");
+            break;
+        case CIP_GET_ATTR_LIST:
+            SCLogDebug("DecodeCIPRequest - CIP_GET_ATTR_LIST\n");
+            break;
+        case CIP_SET_ATTR_LIST:
+            SCLogDebug("DecodeCIPRequest - CIP_SET_ATTR_LIST\n");
+            break;
+        case CIP_RESET:
+            SCLogDebug("DecodeCIPRequest - CIP_RESET\n");
+            break;
+        case CIP_START:
+            SCLogDebug("DecodeCIPRequest - CIP_START\n");
+            break;
+        case CIP_STOP:
+            SCLogDebug("DecodeCIPRequest - CIP_STOP\n");
+            break;
+        case CIP_CREATE:
+            SCLogDebug("DecodeCIPRequest - CIP_CREATE\n");
+            break;
+        case CIP_DELETE:
+            SCLogDebug("DecodeCIPRequest - CIP_DELETE\n");
+            break;
+        case CIP_MSP:
+            SCLogDebug("DecodeCIPRequest - CIP_MSP\n");
+            DecodeCIPRequestMSPPDU(input, input_len, enip_data, offset);
+            break;
+        case CIP_APPLY_ATTR:
+            SCLogDebug("DecodeCIPRequest - CIP_APPLY_ATTR\n");
+            break;
+        case CIP_KICK_TIMER:
+            SCLogDebug("DecodeCIPRequest - CIP_KICK_TIMER\n");
+            break;
+        case CIP_OPEN_CONNECTION:
+            SCLogDebug("DecodeCIPRequest - CIP_OPEN_CONNECTION\n");
+            break;
+        case CIP_CHANGE_START:
+            SCLogDebug("DecodeCIPRequest - CIP_CHANGE_START\n");
+            break;
+        case CIP_GET_STATUS:
+            SCLogDebug("DecodeCIPRequest - CIP_GET_STATUS\n");
+            break;
+        default:
+            SCLogDebug("DecodeCIPRequest - CIP SERVICE 0x%x\n", service);
+    }
+
+    return ret;
+}
+
+
+/**
+ * \brief Deocde CIP Request Path
+ * @param input, input_len data stream
+ * @param enip_data stores data from Packet
+ * @param offset current point in the packet
+ * @param cipserviced the cip service rule
+ * @return 1 Packet matches
+ * @return 0 Packet not match
+ */
+int DecodeCIPRequestPathPDU(uint8_t *input, uint32_t input_len,
+        CIPServiceEntry *node, uint16_t offset)
+{
+
+    //SCLogDebug("DecodeCIPRequestPath: service 0x%x size %d length %d\n",
+    //        node->service, node->request.path_size, input_len);
+
+    if (node->request.path_size < 1)
+    {
+        //SCLogDebug("DecodeCIPRequestPath: empty path or CIP Response\n");
+        return 0;
+    }
+
+    int bytes_remain = node->request.path_size;
```

Comment:
input_len is the total length of the payload.  Bytes_remain is specific to the bytes remaining to be processed in the CIP path